### ISSUE TITLE
ERROR: Could not list test cases of executable (works fine from command line)

### ISSUE THREAD

**Issue by drzoidberg33** [csoltenborn/GoogleTestAdapter#138]
Hey guys,

I'm trying to figure out what's going on with the test discovery and I'm stumped why it won't execute the `--gtest_list_tests` command even though this works perfectly fine when I run it through the command line.

Here is debug output (I've just replaced the project name with a generic name for privacy reasons)

```
------ Discover test started ------
Visual Studio Version: VS2015
Google Test Adapter: Test discovery starting...
Solution settings: AdditionalTestExecutionParam: '', BatchForTestSetup: '', BatchForTestTeardown: '', BreakOnFailure: False, CatchExceptions: True, DebugMode: True, KillProcessesOnCancel: False, MaxNrOfThreads: 16, NrOfTestRepetitions: 1, ParallelTestExecution: False, ParseSymbolInformation: True, PathExtension: '', PrintTestOutput: False, RunDisabledTests: False, ShowReleaseNotes: True, ShuffleTests: False, ShuffleTestsSeed: 0, TestDiscoveryRegex: '', TestDiscoveryTimeoutInSeconds: 30, TestNameSeparator: '', TimestampOutput: False, TraitsRegexesAfter: {}, TraitsRegexesBefore: {}, UseNewTestExecutionFramework: True, WorkingDir: ''
Settings for test executable 'C:\Users\timvd\Development\Project\my-project-name\build\Windows\Win32\Deploy\Debug\My Project Name Tests.exe': AdditionalTestExecutionParam: '', BatchForTestSetup: '', BatchForTestTeardown: '', BreakOnFailure: False, CatchExceptions: True, DebugMode: True, KillProcessesOnCancel: False, MaxNrOfThreads: 16, NrOfTestRepetitions: 1, ParallelTestExecution: False, ParseSymbolInformation: True, PathExtension: '', PrintTestOutput: False, RunDisabledTests: False, ShowReleaseNotes: True, ShuffleTests: False, ShuffleTestsSeed: 0, TestDiscoveryRegex: '', TestDiscoveryTimeoutInSeconds: 30, TestNameSeparator: '', TimestampOutput: False, TraitsRegexesAfter: {}, TraitsRegexesBefore: {}, UseNewTestExecutionFramework: True, WorkingDir: ''
No Google Test indicator file found for executable C:\Users\timvd\Development\Project\my-project-name\build\Windows\Win32\Deploy\Debug\My Project Name Tests.exe
'C:\Users\timvd\Development\Project\my-project-name\build\Windows\Win32\Deploy\Debug\My Project Name Tests.exe' matches default regex '[Tt]est[s]?\.exe'
Settings for test executable 'C:\Users\timvd\Development\Project\my-project-name\build\Windows\Win32\Deploy\Debug\Project Update Service Launcher.exe': AdditionalTestExecutionParam: '', BatchForTestSetup: '', BatchForTestTeardown: '', BreakOnFailure: False, CatchExceptions: True, DebugMode: True, KillProcessesOnCancel: False, MaxNrOfThreads: 16, NrOfTestRepetitions: 1, ParallelTestExecution: False, ParseSymbolInformation: True, PathExtension: '', PrintTestOutput: False, RunDisabledTests: False, ShowReleaseNotes: True, ShuffleTests: False, ShuffleTestsSeed: 0, TestDiscoveryRegex: '', TestDiscoveryTimeoutInSeconds: 30, TestNameSeparator: '', TimestampOutput: False, TraitsRegexesAfter: {}, TraitsRegexesBefore: {}, UseNewTestExecutionFramework: True, WorkingDir: ''
No Google Test indicator file found for executable C:\Users\timvd\Development\Project\my-project-name\build\Windows\Win32\Deploy\Debug\Project Update Service Launcher.exe
'C:\Users\timvd\Development\Project\my-project-name\build\Windows\Win32\Deploy\Debug\Project Update Service Launcher.exe' does not match default regex '[Tt]est[s]?\.exe'
Settings for test executable 'C:\Users\timvd\Development\Project\my-project-name\build\Windows\Win32\Deploy\Debug\Project Update Service.exe': AdditionalTestExecutionParam: '', BatchForTestSetup: '', BatchForTestTeardown: '', BreakOnFailure: False, CatchExceptions: True, DebugMode: True, KillProcessesOnCancel: False, MaxNrOfThreads: 16, NrOfTestRepetitions: 1, ParallelTestExecution: False, ParseSymbolInformation: True, PathExtension: '', PrintTestOutput: False, RunDisabledTests: False, ShowReleaseNotes: True, ShuffleTests: False, ShuffleTestsSeed: 0, TestDiscoveryRegex: '', TestDiscoveryTimeoutInSeconds: 30, TestNameSeparator: '', TimestampOutput: False, TraitsRegexesAfter: {}, TraitsRegexesBefore: {}, UseNewTestExecutionFramework: True, WorkingDir: ''
No Google Test indicator file found for executable C:\Users\timvd\Development\Project\my-project-name\build\Windows\Win32\Deploy\Debug\Project Update Service.exe
'C:\Users\timvd\Development\Project\my-project-name\build\Windows\Win32\Deploy\Debug\Project Update Service.exe' does not match default regex '[Tt]est[s]?\.exe'
Settings for test executable 'C:\Users\timvd\Development\Project\my-project-name\build\Windows\Win32\Deploy\Debug\My Project Name Tests.exe': AdditionalTestExecutionParam: '', BatchForTestSetup: '', BatchForTestTeardown: '', BreakOnFailure: False, CatchExceptions: True, DebugMode: True, KillProcessesOnCancel: False, MaxNrOfThreads: 16, NrOfTestRepetitions: 1, ParallelTestExecution: False, ParseSymbolInformation: True, PathExtension: '', PrintTestOutput: False, RunDisabledTests: False, ShowReleaseNotes: True, ShuffleTests: False, ShuffleTestsSeed: 0, TestDiscoveryRegex: '', TestDiscoveryTimeoutInSeconds: 30, TestNameSeparator: '', TimestampOutput: False, TraitsRegexesAfter: {}, TraitsRegexesBefore: {}, UseNewTestExecutionFramework: True, WorkingDir: ''
Parsing pdb file "C:\Users\timvd\Development\Project\my-project-name\build\Windows\Win32\Build\Debug\My Project Name Tests.pdb"
Found 305 test method symbols and 0 trait symbols in binary C:\Users\timvd\Development\Project\my-project-name\build\Windows\Win32\Deploy\Debug\My Project Name Tests.exe
ERROR: Could not list test cases of executable 'C:\Users\timvd\Development\Project\my-project-name\build\Windows\Win32\Deploy\Debug\My Project Name Tests.exe': executing process failed with return code 20001
Command executed: 'C:\Users\timvd\Development\Project\my-project-name\build\Windows\Win32\Deploy\Debug\My Project Name Tests.exe --gtest_list_tests', working directory: 'C:\Users\timvd\Development\Project\my-project-name\build\Windows\Win32\Deploy\Debug'
Command produced no output
Found 0 tests in executable C:\Users\timvd\Development\Project\my-project-name\build\Windows\Win32\Deploy\Debug\My Project Name Tests.exe
Test discovery completed, overall duration: 00:00:01.7687418
ERROR: 
================
The following errors and warnings occured during test discovery:
ERROR: Could not list test cases of executable 'C:\Users\timvd\Development\Project\my-project-name\build\Windows\Win32\Deploy\Debug\My Project Name Tests.exe': executing process failed with return code 20001
Command executed: 'C:\Users\timvd\Development\Project\my-project-name\build\Windows\Win32\Deploy\Debug\My Project Name Tests.exe --gtest_list_tests', working directory: 'C:\Users\timvd\Development\Project\my-project-name\build\Windows\Win32\Deploy\Debug'
Command produced no output
========== Discover test finished: 0 found (0:00:02,1276578) ==========
```

Example of the output from the command line:

```
C:\Users\timvd\Development\Project\my-project-name\build\Windows\Win32\Deploy\Debug>"My Project Name Tests.exe" --gtest_list_tests
Running main() from gtest_main.cc
TestGroup1.
  Test1
  Test2
  Test3
  Test4
  Test5
  Test6
TestGroup2.
  Test1
```

Any help would be appreciated, thanks!

**Comment by jgefele** [comment]
Hi,

did you see that your Tests.exe returned exit code? Does this have any meaning to you? Maybe your tests try to access any resources / config / whatever which fails?

I can see a very subtle difference in how you call the tests (file name) and how the test adapter calls the tests (full file path). It's unlikely that this causes any trouble but who knows? 

Can you please try this: `C:\Users\timvd\Development\Project\my-project-name\build\Windows\Win32\Deploy\Debug>"C:\Users\timvd\Development\Project\my-project-name\build\Windows\Win32\Deploy\Debug\My Project Name Tests.exe" --gtest_list_tests`

Last thing I can think of: could there be any differences in the environment variables on your command line and those used in the test adapter? Run `set` on your command line to see all environment variables and think about if any variable missing could trigger this error.

You can also try to debug your Tests.exe with:
Command: `C:\Users\timvd\Development\Project\my-project-name\build\Windows\Win32\Deploy\Debug\My Project Name Tests.exe`
Command Arguments: `--gtest-list-tests`
Working Directory: `C:\Users\timvd\Development\Project\my-project-name\build\Windows\Win32\Deploy\Debug`
Environment: empty
Merge Environment: `Yes`

This should mimic how the test adapter calls the Tests.exe. Maybe debugging like that gives you a hint?


Cheers,
Jonas

**Comment by csoltenborn** [comment]
Closed because of inactivity. Feel free to add your comments (e.g. in case our hints were helpful (which one?)) or to re-open if your problem isn't solved.

**Comment by drzoidberg33** [comment]
@csoltenborn Sorry about neglecting this issue - but I seem to have found the problem.

It appears the spaces in the filename of the test executable are causing the problem, I rebuilt and removed the spaces and everything is getting picked up perfectly now.

It would be nice if there was more verbose debug option to see what exactly is being outputted, then I would be able to be 100% sure this was the issue.

I had this same project working on a different machine a while ago and didn't have this issue so I'm really not sure why it happens now.

**Comment by csoltenborn** [comment]
Don't worry, I just made it a habit to close issues if there's no activity for around a week.

Is this maybe related to [this](https://github.com/Microsoft/TestAdapterForGoogleTest/issues/18) weird issue? If so, then it will be fixed in the next version.

Anyway, since you have apparently already used the *Debug mode* option: Where is information lacking? What would you have needed to more easily figure out the cause of your issue?

**Comment by drzoidberg33** [comment]
> Is this maybe related to this weird issue?

Yes seems similar, although my path never had spaces in it but the filename.

> Where is information lacking? 

The output of from the `--gtest-list-tests` command.

Your fix for #18 will likely then fix this so thanks!
